### PR TITLE
Show category / tag in Expense preview view

### PR DESCRIPTION
**Pull request by mkzie2**
### Fixed Issues

$ https://github.com/Expensify/App/issues/52261
PROPOSAL: https://github.com/Expensify/App/issues/52261#issuecomment-2465507832

### Tests

1. Submit some expenses in a workspace chat with category/tag:
- No category/tag
- Short/long category, no tag
- No category, short/long tag
- Short/long category, short/long tag
2. Open the expense preview
3. Verify that all cases, category/tag is displayed with the correct style
- [x] Verify that no errors appear in the JS console

### Offline tests

Same
### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."
1. Submit some expenses in a workspace chat with category/tag:
- No category/tag
- Short/long category, no tag
- No category, short/long tag
- Short/long category, short/long tag
2. Open the expense preview
3. Verify that If the category is short, the tag will take the remaining width
4. Verify that If the category is too long, the category will take 50% width and is truncated with `...`
5. Verify that If both the category and the tag are too long, both of them take the same width and are truncated with `...`
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/710c3712-279e-4381-a483-1d18aa3ccb63

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/7e7b665e-9af3-4c00-9a7b-3949c671d7fc

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/17886254-a5bf-4938-872c-ce9fcaf6233a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/9a7cb27e-6677-4da3-a144-205e840efe3b

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/39a977ed-a7a3-408f-b43b-9536a8db775b

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/9f57a824-5f4e-4a01-ae24-7072e846f9c2

</details>

### CODE REVIEW

**Comment by dannymcclain** [comment]
Screenshots are looking good! Going to try to run a test build and get @Expensify/design's eyes on this as well.

**Comment by dannymcclain** [comment]
😭 Oof yes, I agree we need to deal with that. Nothing is immediately coming to mind, but I agree we need to clean it up somehow.

**Comment by dubielzyk-expensify** [comment]
I wonder if we should at least make these the same font size? Feels like there's too many sizes going on at the moment. Given they use the same icon size as well. Thoughts?
CleanShot 2024-11-21 at 13 19 28@2x

**Comment by mkzie2** [comment]
@dannymcclain I updated your suggestion here. Here is the latest result.


Screenshot 2024-11-21 at 11 52 00

**Comment by shawnborton** [comment]
> Feels like there's too many sizes going on at the moment. Given they use the same icon size as well. Thoughts?

Totally agree, though it seems like we can actually solve this one by just moving the pending/scanning text to the top eye brow line.

**Comment by shawnborton** [comment]
Spun up another test build for testing, cc @dannymcclain - let us know if that matches your expectations re: design styles!

**Comment by shawnborton** [comment]
The one thing that stands out to me is the font size. I think you are using micro (11px) in your mocks, but I think the build is using label (13px)? Let's get some clarification there.

**Comment by dannymcclain** [comment]
>The one thing that stands out to me is the font size. I think you are using micro (11px) in your mocks, but I think the build is using label (13px)? Let's get some clarification there.

Good catch! Yes, the mocks use Micro (11px) and it does indeed look like the build is using Label (13px). Let's get that updated.

Also it looks like expense previews without categories and tags are getting some extra bottom padding in the build?

| This build | Desktop staging |
|--------|--------|
| CleanShot 2024-11-22 at 09 04 36@2x | CleanShot 2024-11-22 at 09 05 28@2x |
| CleanShot 2024-11-22 at 09 06 16@2x | CleanShot 2024-11-22 at 09 06 55@2x |

**Comment by shawnborton** [comment]
Good catches Danny!

**Comment by mkzie2** [comment]
@dannymcclain Fixed two bugs above.

**Comment by shawnborton** [comment]
Will spin up a new test build.

**Comment by dannymcclain** [comment]
That new test build is looking good! (After I opened the build link I had to refresh to get the latest styles just FYI Shawn)

**Comment by shawnborton** [comment]
Nice, yeah that feels good to me. Let's get it into final review! People are gonna love this one.

**Comment by dubielzyk-expensify** [comment]
Just wanted to confirm that we're wanting to reserve the space to be 50% width on category and tags right? 
CleanShot 2024-11-25 at 15 09 10@2x

Also this is super minor, but is it possible to do trim before the ellipsis? You can see in the screenshot above that we have a space before the `...` which looks a tad odd.

**Comment by shawnborton** [comment]
> Just wanted to confirm that we're wanting to reserve the space to be 50% width on category and tags right?

My understanding is that this is probably the easiest way to do it given the limits of flexbox. Ideally we'd just use all of the available space to us in that horizontal area but I think it's tricky to have say the category take up like 70% of the space and the tag take up the rest. The 50/50 split makes it easier to force the truncation I think. 

> You can see in the screenshot above that we have a space before the ... which looks a tad odd.

Agree, good catch!

**Comment by dannymcclain** [comment]
Re: Spacing, I'm down to just go with 50% if it drastically simplifies the implementation, though do you both agree that this would be ideal?

CleanShot 2024-11-25 at 08 30 10@2x

I'm not sure how technically feasible the above is, so again, I'm happy to just go with 50/50. 👍

**Comment by shawnborton** [comment]
> though do you both agree that this would be ideal?

100% agree that would be ideal! But yeah, same page - just not sure how easy it is to pull off. @mkzie2 thoughts?

**Comment by mkzie2** [comment]
@shawnborton So we always want to force the width 50/50, right? or only category/tag is too long

**Comment by mkzie2** [comment]
> You can see in the screenshot above that we have a space before the ... which looks a tad odd.

I think it's only a special case when the last character before `...` is space. It's hard to detect what is the last character before `...`

**Comment by shawnborton** [comment]
> So we always want to force the width 50/50, right? or only category/tag is too long

I think we would love it if [Danny's mockups above](https://github.com/Expensify/App/pull/52758) were possible. But if not, we understand why we might need to do 50/50.

**Comment by mkzie2** [comment]
@shawnborton I just tried to do that but the problem here is we cannot know whether the category should be truncated. Or if we make the category take the remaining width if the category is longer than the tag, the UI doesn't correct when both the tag and category are short. So the best way here is to do 50/50 and the tag will always take the remaining width.

**Comment by shawnborton** [comment]
Yeah, I had a feeling. Let's make sure we have good screenshots for this that cover a variety of scenarios: long category + short tag, short category + short tag, short category + long tag, long category + long tag, etc

**Comment by dubielzyk-expensify** [comment]
> So the best way here is to do 50/50 and the tag will always take the remaining width.

Thanks for giving it a crack.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-11-27 at 11 53 23
Screenshot 2024-11-27 at 11 53 33
Screenshot 2024-11-27 at 11 53 55
Screenshot 2024-11-27 at 11 54 09

</details>



<details>
<summary>iOS: Native</summary>

Screenshot 2024-11-27 at 11 46 47
Screenshot 2024-11-27 at 11 47 03
Screenshot 2024-11-27 at 11 47 13
Screenshot 2024-11-27 at 11 47 29
Screenshot 2024-11-27 at 11 47 34

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2024-11-27 at 11 58 32
Screenshot 2024-11-27 at 11 58 38
Screenshot 2024-11-27 at 11 58 51
Screenshot 2024-11-27 at 11 58 58
Screenshot 2024-11-27 at 11 59 07

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


Screenshot 2024-11-27 at 11 15 15
Screenshot 2024-11-27 at 11 16 12
Screenshot 2024-11-27 at 11 17 03
Screenshot 2024-11-27 at 11 34 02
Screenshot 2024-11-27 at 11 34 11

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-11-27 at 11 51 23
Screenshot 2024-11-27 at 11 51 34
Screenshot 2024-11-27 at 11 51 39

</details>

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by Beamanator** [comment]
@mkzie2 would you please update the QA steps so that they're perfectly clear what QA should be verifying?

> Verify that all cases, category/tag is displayed with the correct style

Specifically there ^ I think will be hard for QA to know exactly what the "correct style" is unless you write it out 🙏  Thanks!

**Review by Beamanator — APPROVED** [review]
Lookin great! LGTM

**Comment by Beamanator** [comment]
Also @shawnborton @dannymcclain thanks for the design consultations!! 🙏 🚀

**Comment by mkzie2** [comment]
@Beamanator Updated the QA test step.

**Comment by Beamanator** [comment]
Thanks very much!

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx [hunk 1]
```diff
@@ -427,6 +430,43 @@ function MoneyRequestPreviewContent({
...
+                                                    />
+                                                    <Text
+                                                        numberOfLines={1}
+                                                        style={[styles.textMicroSupporting, styles.pre, styles.flexShrink1]}
+                                                    >
+                                                        {category}
+                                                    </Text>
+                                                </View>
+                                            )}
+                                            {!!tag && (
```
Comment (jayeshmangwani) [inline src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:452]: coming from this issue https://github.com/Expensify/App/issues/53454, here we’re displaying the tag name if tag value is available. However, this alone is not sufficient for display in the preview. Rooms created under a workspace with dependent tags enabled will also show the tags in the preview. That said, tags should only appear in the expense preview if the report is a policy expense chat. More details can be found here: https://github.com/Expensify/App/issues/53454#issuecomment-2514595733

**Pull request merged**